src/sinks/databend/config.rs

```diff
@@ -48,11 +45,11 @@ pub struct DatabendConfig {
     pub database: String,
 
     #[configurable(derived)]
-    #[serde(
-        default,
-        skip_serializing_if = "crate::serde::skip_serializing_if_default"
-    )]
-    pub encoding: Transformer,
+    pub encoding: EncodingConfig,
```

Comment:
Similar to @zamazan4ik's suggestion around `compression`, I think we should have a specific `DatabendEncodingConfig` that could "wrap" `EncodingConfig` by adhering to this [implementation](https://cs.github.com/vectordotdev/vector/blob/ab678631a7bfc64db5dccf09de805c83b7ee587b/src/codecs/encoding/config.rs#L45-L55) while only exposing supported codecs.

I haven't spiked that code out, so it may be trickier than I anticipated.

Reply:
Let me have a try.